glambie/util/timeseries_helpers.py

```diff
@@ -0,0 +1,358 @@
+        the x-coordinates of the input data
+    y : np.array, optional
+        the y-coordinates of the input data, by default None
+    clip : bool, optional
+        limit the output data to values for which enough data exists to form a complete average,
+        returned is a np.array of the same size as y with nan values inserted for the clip, by default False
+
+    Returns
+    -------
+    np.array
+        the averaged y-values, has same length as y
+    """
+    if y is not None:
+        n = len(x)
+
+        ry = np.empty(n,)
+        ry.fill(np.NAN)
+
+        for i in range(n):
+            ok = np.logical_and(
+                x > x[i] - dx / 2.,
+                x < x[i] + dx / 2.)
+            if ok.any():
+                ry[i] = np.mean(y[ok])
+        if clip:
+            ok = np.logical_or(
+                x < np.min(x) + dx / 2 - 1,
+                x > np.max(x) - dx / 2 + 1)
+            if ok.any():
+                ry[ok] = np.NAN
+        return ry
+    else:
+        result = np.cumsum(x, dtype=x.dtype)
+        result[dx:] = result[dx:] - result[:-dx]
+        return result[dx - 1:] / dx
+
+
+def timeseries_as_months(fractional_year_array: np.array, downsample_to_month: bool = True) -> np.array(float):
+    """Calculates an array (in units of fractional years) of evenly spaced monthly intervals over the timespan of
+    a given date array (in units of fractional years)
+
+    This function can be used as input for resampling to ensure that timesteps of different time series are consistent
+
+    Note that the year is evenly split into 12 parts within this function, meaning that not each month starts at
+    the first of the month.
+
+    Parameters
+    ----------
+    fractional_year_array : np.array
+        The input time-series
+    downsample_to_month : bool, optional
+        if True, extends the series to contain values each month
+        if False, only existing dates are shifted to nearest month
+
+    Returns
+    -------
+    np.array[float]
+        the new monthly array of fractional years
+    """
+    if downsample_to_month:
+        t0 = math.floor(np.min(fractional_year_array) * 12) / 12.
+        t1 = math.ceil(np.max(fractional_year_array) * 12 + 1) / 12.
+        monthly_array = np.arange((t1 - t0) * 12) / 12. + t0
+    else:
+        monthly_array = np.floor(fractional_year_array * 12) / 12.
+    return monthly_array
+
+
+def combine_timeseries(t_array: list[np.ndarray],
+                       y_array: list[np.ndarray],
+                       outlier_tolerance: float = None,
+                       calculate_as_errors: bool = False,
+                       perform_moving_average: bool = False,
+                       verbose=False) \
+        -> Union[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Combines a number of input timeseries sequences
+
+    Parameters
+    ----------
+    t_array : list[np.ndarray]
+        a list/iterable of the time-series arrays to be combined
```

Comment:
I think here, y_array would be called the timeseries and t_array would be called the time axis. Again, though, this is a minor issue and the current solution is fine.